src/views/upload.tsx

```diff
@@ -4,24 +4,28 @@ import { useGetUserSettings } from "ui/hooks/settings";
 import BlankScreen, { LoadingScreen } from "ui/components/shared/BlankScreen";
 import UploadScreen from "ui/components/UploadScreen";
 
-function UploadScreenWrapper() {
+function UploadScreenWrapper({ onUpload }: { onUpload: () => void }) {
   const recordingId = useGetRecordingId();
-  const { recording } = useGetRecording(recordingId);
+  const { recording, refetch } = useGetRecording(recordingId);
   // Make sure to get the user's settings before showing the upload screen.
   const { userSettings, loading } = useGetUserSettings();
 
   useEffect(() => {
     if (recording?.isInitialized) {
       window.onbeforeunload = null;
-      document.location.reload();
```

Comment:
I'm not 100% sure, but will `recording.isInitialized` ever be `true` before we call `refetch()`?

If it is true, that means that we've updated the recording in the cache. Do we need to still call refetch then?

Reply:
Just tested it and seems to work. 😳